File: April/Dhruvabhat24/C++/April 11/Solution.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        vector<char>stk;
        string ans;
        for(char c :num){
            while(k>0 && !stk.empty() && stk.back()>c){
                stk.pop_back();
                k--;
            }
            stk.push_back(c);
        }
        while(k>0){
            stk.pop_back();
            k--;
        }
        for(char c : stk){
            if(ans.empty() && c=='0') continue;
            ans.push_back(c);

        }
        return ans.empty()?"0":ans;
    }
};

File: April/Dhruvabhat24/Java/April 11/Solution.java
class Solution {
    public String removeKdigits(String num, int k) {
        Stack<Character> stk = new Stack<>();
        StringBuilder ans = new StringBuilder();
        for (char c : num.toCharArray()) {
            while (k > 0 && !stk.isEmpty() && stk.peek() > c) {
                stk.pop();
                k--;
            }
            stk.push(c);
        }
        while (k > 0) {
            stk.pop();
            k--;
        }
        for (char c : stk) {
            if (ans.length() == 0 && c == '0') continue;
            ans.append(c);
        }
        return ans.length() == 0 ? "0" : ans.toString();
    }
}
